src/v3/src/transformer/terminal/transformTerminalTransaction.ts

```diff
@@ -116,7 +116,23 @@ const appendViewLinks = (
       && isCancelAvailable) {
     const backToSigninUri = getBackToSignInUri(widgetProps);
     if (backToSigninUri) {
+      // If a backtoSigninUri is set, always use it
       cancelLink.options.href = backToSigninUri;
+    } else if (containsMessageKey(TERMINAL_KEY.VERIFICATION_TIMED_OUT, transaction.messages)) {
+      // This covers terminal views that should have a page reload to start a new transaction in the case
+      // that the URI is specifing a particular application to login to, so we need to stay on the same URI
```

Comment:
[super nit (only if you're planning on addressing comments)] 
```suggestion
      // that the URI is specifying a particular application to login to, so we need to stay on the same URI
```